test/langtools/tools/javac/modules/AutomaticModules.java

```diff
@@ -665,7 +665,31 @@ public void testAutomaticModuleNameCorrect(Path base) throws Exception {
                          "-XDrawDiagnostics")
                 .outdir(classes)
                 .files(findJavaFiles(src))
-                .run(Task.Expect.SUCCESS)
+                .run(Task.Expect.SUCCESS,
+                     // Processor verifies api.Api is enclosed by an automatic module.
+                     new javax.annotation.processing.AbstractProcessor() {
+                         @Override
+                         public java.util.Set<String> getSupportedAnnotationTypes() {
+                             return java.util.Set.of("*");
+                         }
+
+                         @Override
+                         public  javax.lang.model.SourceVersion getSupportedSourceVersion() {
+                             return javax.lang.model.SourceVersion.latestSupported();
+                         }
+
+                         @Override
+                         public boolean process(java.util.Set<? extends javax.lang.model.element.TypeElement> annotations,
+                                                    javax.annotation.processing.RoundEnvironment roundEnv) {
+                             if (!roundEnv.processingOver()) {
+                                 javax.lang.model.util.Elements elts = processingEnv.getElementUtils();
+                                 if (!elts.isAutomaticModule(elts.getModuleOf(elts.getTypeElement("api.Api")))) {
+                                     throw new RuntimeException("module of class Api is not automatic");
+                                 }
+                             }
+                             return true;
+                         }
+                     })
                 .writeAll()
                 .getOutputLines(Task.OutputKind.DIRECT);
```

Comment:
I know this code predates your patch but now that you are here :) these two lines are unnecessary as the output is not kept. I mean you can remove:
           .writeAll()
           .getOutputLines(Task.OutputKind.DIRECT)